src/Extension/TaskList/TaskListItemMarkerParser.php

```diff
@@ -40,18 +38,15 @@ public function parse(InlineParserContext $inlineContext): bool
         $cursor   = $inlineContext->getCursor();
         $oldState = $cursor->saveState();
 
-        $m = $cursor->match('/\[[ xX]\]/');
-        if ($m === null) {
-            return false;
-        }
+        $cursor->advanceBy(\mb_strlen($match));
```

Comment:
Can be replaced by `$cursor->advanceBy(3);` since we'll always match on exactly 3 characters